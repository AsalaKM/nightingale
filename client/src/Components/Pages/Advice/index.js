import React, { Component } from "react";
import Collapsible from "react-collapsible";

// styling
import { Wrapper, Article } from "./index.style";

// common components
import Title from "../../Common/Title";

export default class Advice extends Component {
  render() {
    return (
      <React.Fragment>
        <Title value="Advice" />
        <Wrapper>
          <Collapsible
            trigger={<h3>Section One</h3>}
            transitionTime={300}
            easing="ease"
            openedClassName="open"
          >
            <Article>
              <p>Collapsible content 1</p>
              <a href="www.article.com" target="_blank">
                Read more
              </a>
            </Article>
            <Article>
              <p>Collapsible content 1</p>
              <a href="www.article.com" target="_blank">
                Read more
              </a>
            </Article>
          </Collapsible>
          <Collapsible
            trigger={<h3>Section Two</h3>}
            transitionTime={300}
            easing="ease"
          >
            <Article>
              <p>Collapsible content 1</p>
              <a href="www.article.com" target="_blank">
                Read more
              </a>
            </Article>
          </Collapsible>
          <Collapsible
            trigger={<h3>Section Three</h3>}
            transitionTime={300}
            easing="ease"
          >
            <Article>
              <p>Collapsible content 1</p>
              <a href="www.article.com" target="_blank">
                Read more
              </a>
            </Article>
            <Article>
              <p>Collapsible content 1</p>
              <a href="www.article.com" target="_blank">
                Read more
              </a>
            </Article>
            <Article>
              <p>Collapsible content 1</p>
              <a href="www.article.com" target="_blank">
                Read more
              </a>
            </Article>
          </Collapsible>
          <Collapsible
            trigger={<h3>Section Four</h3>}
            transitionTime={300}
            easing="ease"
          >
            <Article>
              <p>Collapsible content 1</p>
              <a href="www.article.com" target="_blank">
                Read more
              </a>
            </Article>
            <Article>
              <p>Collapsible content 1</p>
              <a href="www.article.com" target="_blank">
                Read more
              </a>
            </Article>
          </Collapsible>
        </Wrapper>
      </React.Fragment>
    );
  }
}
