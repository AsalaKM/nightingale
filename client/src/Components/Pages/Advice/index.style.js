import styled from "styled-components";

export const Wrapper = styled.div`
  margin-top: 2rem;
  .Collapsible {
    border-top: 1px solid #cccc;
  }

  .Collapsible__trigger {
    position: relative;
    cursor: pointer;
    display: flex;
    padding: 1rem;
  }

  .is-closed {
    :after {
      font-family: "Font Awesome 5 Free";
      content: "\f107";
      position: absolute;
      right: 1rem;
      top: 1rem;
      display: block;
      color: #cccc;
      font-weight: 900;
      font-size: 1.5rem;
      transition: transform 300ms ease;
    }
  }

  .is-open {
    :after {
      font-family: "Font Awesome 5 Free";
      content: "\f107";
      position: absolute;
      right: 1rem;
      top: 1rem;
      display: block;
      color: #cccc;
      font-weight: 900;
      font-size: 1.5rem;
      transition: transform 300ms ease;
      transform: rotateZ(180deg);
    }
  }

  .Collapsible__contentInner {
    padding-left: 1rem;
    padding-right: 1rem;
  }
`;

export const Article = styled.div.attrs({
  className: "flex items-center justify-center"
})`
  margin-bottom: 1rem;
  border-bottom: 1px solid #edeeee;
  padding: 1rem 1rem 0.5rem 1rem;
  transition: all 300ms ease;

  a {
    text-decoration: none;
    cursor: pointer;
    color: #925de5;
  }

  :hover {
    margin-left: 1.5rem;
  }
`;
