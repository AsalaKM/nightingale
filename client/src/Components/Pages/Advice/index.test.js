import React from "react";
import Advice from "./index";
import renderer from "react-test-renderer";

describe("Advice", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<Advice />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
