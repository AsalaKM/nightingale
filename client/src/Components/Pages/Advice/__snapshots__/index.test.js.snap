// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Advice renders correctly 1`] = `
Array [
  <div
    className="sc-htpNat dEQUjC"
  >
    <h1
      className="sc-bxivhb kFbrlN"
    >
      Advice
    </h1>
  </div>,
  <div
    className="sc-bdVaJa cuaDBM"
  >
    <div
      className="Collapsible"
    >
      <span
        className="Collapsible__trigger is-closed"
        onClick={[Function]}
        onKeyPress={[Function]}
        style={null}
        tabIndex={null}
      >
        <h3>
          Section One
        </h3>
      </span>
      <div
        className="Collapsible__contentOuter"
        onTransitionEnd={[Function]}
        style={
          Object {
            "WebkitTransition": "height 300ms ease",
            "height": 0,
            "msTransition": "height 300ms ease",
            "overflow": "hidden",
            "transition": "height 300ms ease",
          }
        }
      >
        <div
          className="Collapsible__contentInner"
        >
          <div
            className="sc-bwzfXH flex items-center justify-center btEvd"
          >
            <p>
              Collapsible content 1
            </p>
            <a
              href="www.article.com"
              target="_blank"
            >
              Read more
            </a>
          </div>
          <div
            className="sc-bwzfXH flex items-center justify-center btEvd"
          >
            <p>
              Collapsible content 1
            </p>
            <a
              href="www.article.com"
              target="_blank"
            >
              Read more
            </a>
          </div>
        </div>
      </div>
    </div>
    <div
      className="Collapsible"
    >
      <span
        className="Collapsible__trigger is-closed"
        onClick={[Function]}
        onKeyPress={[Function]}
        style={null}
        tabIndex={null}
      >
        <h3>
          Section Two
        </h3>
      </span>
      <div
        className="Collapsible__contentOuter"
        onTransitionEnd={[Function]}
        style={
          Object {
            "WebkitTransition": "height 300ms ease",
            "height": 0,
            "msTransition": "height 300ms ease",
            "overflow": "hidden",
            "transition": "height 300ms ease",
          }
        }
      >
        <div
          className="Collapsible__contentInner"
        >
          <div
            className="sc-bwzfXH flex items-center justify-center btEvd"
          >
            <p>
              Collapsible content 1
            </p>
            <a
              href="www.article.com"
              target="_blank"
            >
              Read more
            </a>
          </div>
        </div>
      </div>
    </div>
    <div
      className="Collapsible"
    >
      <span
        className="Collapsible__trigger is-closed"
        onClick={[Function]}
        onKeyPress={[Function]}
        style={null}
        tabIndex={null}
      >
        <h3>
          Section Three
        </h3>
      </span>
      <div
        className="Collapsible__contentOuter"
        onTransitionEnd={[Function]}
        style={
          Object {
            "WebkitTransition": "height 300ms ease",
            "height": 0,
            "msTransition": "height 300ms ease",
            "overflow": "hidden",
            "transition": "height 300ms ease",
          }
        }
      >
        <div
          className="Collapsible__contentInner"
        >
          <div
            className="sc-bwzfXH flex items-center justify-center btEvd"
          >
            <p>
              Collapsible content 1
            </p>
            <a
              href="www.article.com"
              target="_blank"
            >
              Read more
            </a>
          </div>
          <div
            className="sc-bwzfXH flex items-center justify-center btEvd"
          >
            <p>
              Collapsible content 1
            </p>
            <a
              href="www.article.com"
              target="_blank"
            >
              Read more
            </a>
          </div>
          <div
            className="sc-bwzfXH flex items-center justify-center btEvd"
          >
            <p>
              Collapsible content 1
            </p>
            <a
              href="www.article.com"
              target="_blank"
            >
              Read more
            </a>
          </div>
        </div>
      </div>
    </div>
    <div
      className="Collapsible"
    >
      <span
        className="Collapsible__trigger is-closed"
        onClick={[Function]}
        onKeyPress={[Function]}
        style={null}
        tabIndex={null}
      >
        <h3>
          Section Four
        </h3>
      </span>
      <div
        className="Collapsible__contentOuter"
        onTransitionEnd={[Function]}
        style={
          Object {
            "WebkitTransition": "height 300ms ease",
            "height": 0,
            "msTransition": "height 300ms ease",
            "overflow": "hidden",
            "transition": "height 300ms ease",
          }
        }
      >
        <div
          className="Collapsible__contentInner"
        >
          <div
            className="sc-bwzfXH flex items-center justify-center btEvd"
          >
            <p>
              Collapsible content 1
            </p>
            <a
              href="www.article.com"
              target="_blank"
            >
              Read more
            </a>
          </div>
          <div
            className="sc-bwzfXH flex items-center justify-center btEvd"
          >
            <p>
              Collapsible content 1
            </p>
            <a
              href="www.article.com"
              target="_blank"
            >
              Read more
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;
